ppa/archive/views.py

```diff
@@ -247,7 +253,56 @@ class DigitizedWorkDetailView(DetailView):
     model = DigitizedWork
     slug_field = 'source_id'
     slug_url_kwarg = 'source_id'
+    paginate_by = 50
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        # pull in the query if it exists to use
+        query = self.request.GET.get('query', '')
+
+        solr_pageq = None
+        if query:
+            context['query'] = query
+            digwork = context['object']
+            solr_q = 'text:(%s) AND srcid:("%s") AND item_type:(page)' \
+                % (query, digwork.source_id)
+            solr_opts = {
+                'q': solr_q,
+                'sort': 'order asc',
+                'fl': '*',
+                'hl': True,
+                'hl.fl': 'content',
+                # No way to set this to actual number in one search
+                # therefore using an arbitrarily high but not system breaking
+                # number.
+                'hl.snippets': 10000,
+                # not default but recommended
+                'hl.method': 'unified',
+                # default to paginating by 50 rows
+                'rows': 50,
+            }
+
+            logger.info("Solr page keyword search query: %s" % solr_q)
+
+            try:
+                # configure paginator and set in context
+                solr_pageq = PagedSolrQuery(solr_opts)
+                paginator = Paginator(solr_pageq, per_page=self.paginate_by)
+                page = self.request.GET.get('page', 1)
+                context['page_obj'] = paginator.page(page)
+                # get highlights and set in context
+                highlights = solr_pageq.get_results() if solr_pageq else []
+                context['page_highlights'] = highlights
```

Comment:
Ah, I see.  You also need to call `get_highlighting` to get the highlight snippet results returned by Solr.  So here, `highlights` is actually your matching pages and you're not passing in the highlight information to the template for display.